rustls/src/hash_hs.rs

```diff
@@ -58,6 +58,16 @@ impl HandshakeHashBuffer {
         ctx.finish()
     }
 
+    /// Fork the hash computation, producing another context that has the
+    /// same prefix as this one.
+    #[allow(dead_code)] // TODO(@cpu): remove in subsequent commit.
+    pub(crate) fn get_fork(&self) -> Self {
```

Comment:
> Nit: get_() prefixes (also for HandshakeHash) -- maybe just fork() or forked()

Good call, we already use `fork` and `fork_finish` in the `crypto/hash.rs` interface so I think we should match here.

>  is there a reason to not just use Clone?

We could do that for `HandshakeHashBuffer` used before a hash algorithm is decided where the struct just holds `Vec<u8>` and `bool`, but we can't for `HandshakeHash` where we hold a `&'static dyn hash::Hash` that we need to call `.fork()` on. I think consistency between `HandshakeHashBuffer` and `HandshakeHash` is a good reason to avoid using `clone()` for one and `fork()` for the other

Reply:
Hmm, I don't know. How is `fork()` really different conceptually from `clone()`? Could do a custom `Clone` impl that uses `fork()` to make it work.

Reply:
>  Could do a custom Clone impl that uses fork() to make it work.

That's a good point. I like that idea :+1:

Reply:
Fixed up in https://github.com/rustls/rustls/pull/1739